plugins/auth-backend-module-vmware-cloud-provider/src/resolvers.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import {
+  createSignInResolverFactory,
+  OAuthAuthenticatorResult,
+  PassportProfile,
+  SignInInfo,
+} from '@backstage/plugin-auth-node';
+
+/**
+ * Available sign-in resolvers for the VMware Cloud auth provider.
+ *
+ * @public
+ */
+export namespace vmwareCloudSignInResolvers {
+  /**
+   * Looks up the user by matching their profile email to the entity's profile email.
+   * If that fails, sign in the user without associating with a catalog user.
+   */
+  export const profileEmailMatchingUserEntityEmail =
+    createSignInResolverFactory({
+      create() {
+        return async (
+          info: SignInInfo<OAuthAuthenticatorResult<PassportProfile>>,
+          ctx,
+        ) => {
+          const email = info.profile.email;
+
+          if (!email) {
+            throw new Error(
+              'VMware login failed, user profile does not contain an email',
+            );
+          }
+
+          const userEntityRef = stringifyEntityRef({
+            kind: 'User',
+            name: email,
```

Comment:
noting that this is the full email in the name - requires catalog validation rules tweak to work

Reply:
Uh should we strip the email part? it would be best if this works out of the box, it's just that this is what is in use in VMware's Backstage project...

Reply:
Honestly it would probably be best to make at least at-signs valid name characters out of the box... But our default rules supposedly follow the k8s object naming rules

Reply:
I'd rather use one of the `commonSignInResolvers` here instead of maintaining this one in the open fwiw:

https://github.com/backstage/backstage/blob/e84ad774c612dc661250786d4d5dec5fbe57f1e3/plugins/auth-node/src/sign-in/commonSignInResolvers.ts#L50-L72

Reply:
I think its feasible, the part that actually matter was the profile transformer which grabbed the profile info from VMware's response, @jamieklassen maybe log this as an issue to do a bit latter?